oshmem/include/pshmemx.h

```diff
@@ -16,110 +16,194 @@
 extern "C" {
 #endif
 
+
+/*
+ * Legacy API
+ * old init/destruct functions - not in the open shmem spec but still supported
+ */
+OSHMEM_DECLSPEC  void pstart_pes(int npes);
+
+OSHMEM_DECLSPEC  int p_num_pes(void);
+OSHMEM_DECLSPEC  int p_my_pe(void);
+
+OSHMEM_DECLSPEC  void* pshmalloc(size_t size);
+OSHMEM_DECLSPEC  void* pshmemalign(size_t align, size_t size);
+OSHMEM_DECLSPEC  void* pshrealloc(void *ptr, size_t size);
+OSHMEM_DECLSPEC  void pshfree(void* ptr);
+
+OSHMEM_DECLSPEC  void pshmem_char_put(char *target, const char *source, size_t len, int pe);
+OSHMEM_DECLSPEC  void pshmem_char_get(char *target, const char *source, size_t len, int pe);
```

Comment:
I believe that it is a good point but this capability requires changes that probably too heavy if follow right way.
We need to introduce new configure option as oshmem-interface-warning and make it consistent with mpi-interface-warning, duplicate mpi_portable_platform.h file to recogize a compiler and include it into shmem.h header. Define new or use OMPI_BUILDING inside shmem.h etc. I am not sure if we can include additional files inside shmem.h.
So I do not have clear point on this proposal.
@miked-mellanox what do you think?